parlai/core/torch_ranker_agent.py

```diff
@@ -280,6 +287,34 @@ def _build_candidates(self, batch, source, mode):
             cand_vecs = label_vecs
             label_inds = label_vecs.new_tensor(range(batchsize))
 
+        elif source == 'batch-all-cands':
+            warn_once(
+                '[ Executing {} mode with all candidates provided in the batch ]'
+                ''.format(mode)
+            )
+            if batch.candidate_vecs is None:
+                raise ValueError(
+                    "If using candidate source 'batch-all-cands', then batch."
+                    "candidate_vecs cannot be None. If your task does not have "
+                    "inline candidates, consider using one of "
+                    "--{m}={{'batch','fixed','vocab'}}."
```

Comment:
nit: can't use "batch" if candidate vecs is None

Reply:
With 'batch' only labels are considered.